.github/workflows/build-rock.yaml

```diff
@@ -0,0 +1,76 @@
+name: Build
+
+on:
+  workflow_call:
+    inputs:
+      rock-dir:
+        description: "Path to rock directory, i.e. directory containing rockcraft.yaml"
+        required: true
+        type: string
+      source-branch:
+        description: Github branch to checkout. If blank, it will use default values.
+        default: '' # For default behaviour, see https://github.com/actions/checkout/tree/v4/#usage
+        required: false
+        type: string
+    outputs:
+        path-to-rock-file:
+          description: "Path to .rock file produced"
+          value: ${{ jobs.build-rock.outputs.path-to-rock-file}}
+        rock-filename:
+          description: "Filename of the .rock file produced"
+          value: ${{ jobs.build-rock.outputs.rock-filename}}
+        rock-name:
+          description: "Field name of the rockcraft.yaml file built"
+          value: ${{ jobs.build-rock.outputs.rock-name}}
+        rock-version:
+          description: "Field version of the rockcraft.yaml file built"
+          value: ${{ jobs.build-rock.outputs.rock-version}}
+
+jobs:
+  build-rock:
+    name: Build ${{ inputs.rock-dir }}
+    runs-on: ubuntu-22.04
+    outputs:
+      path-to-rock-file: ${{ steps.rockcraft.outputs.rock }}
+      rock-filename: ${{ steps.metadata.outputs.rock-filename }}
```

Comment:
iirc from our conversation yesterday, @kimwnasptd proposed to rename this so sth like `oci-artifact`